src/core/rtos/Queue.cpp

```diff
@@ -0,0 +1,86 @@
+
+#include <core/rtos/Enums.hpp>
+#include <core/rtos/Queue.hpp>
+
+namespace core::rtos {
+
+// TODO: uncomment txNotifyFunction when it is fixed
+Queue::Queue(char* name, uint32_t messageSize, uint32_t numMessages)
+    : name(name), messageSize(messageSize), queueSize(messageSize * numMessages) /*,
+         txNotifyFunction(txNotifyFunctionTemplate<this, TX_QUEUE>)*/
+{}
+
+TXError Queue::init(BytePoolBase& pool) {
+    void* poolPointer;
+    // allocate memory on the pool
+    uint32_t errorCode = pool.allocateMemory(queueSize, NO_WAIT, &poolPointer);
```

Comment:
Not quite. Setting this to NO_WAIT means that if the BytePool is full, the calling thread will not wait for the pool to empty it will just return immediately with an error code. People should be calibrating the size of their BytePools to make sure they are never too full, and generally we don't want to wait on the pool emptying because the pool is likely to never empty, as we generally do not deallocate memory off the BytePool, we only allocate onto it.